doc: update release procedure (#89)
pgroonga/pgroonga.github.io: development/release.md

Use the following enviroment values.

* `GROONGA_REPOSITORY`

  Specify a path for a latest Groonga repository.

>   If you have not cloned the Groonga repository.

Author: 単純に誤記でした。
ここの内容自体の見直しとともに修正しました。

  ```console
  $ mkdir -p ~/work/groonga
  $ git clone --recursive git@github.com:groonga/groonga.git ~/work/groonga/groonga.clean
  $ export GROONGA_REPOSITORY=$HOME/work/groonga/groonga.clean
  ```

>   If you have already cloned the Groonga repository, execute `git pull` beforehand.

Reviewer: In my opinion, we should clone the Groonga repository every release.
Because we prevent  of missing update for the Groonga repository.

Author: 毎回cloneするような記述内容&スクリプト例に変更しました。

* [GitHub Actions][github-actions-pgroonga]

## Check whether we can build packages on `nightly` repository of launchpad.net

For Ubuntu, packages are provided by PPA on launchpad.net.

We have [nightly][launchpad-groonga-nightly] and [ppa][launchpad-groonga-nightly] repositories on launchpad.net.
`nightly` is for testing and `ppa` is for distributing.

We should test whether we can build packages for Ubuntu on the `nightly` repository before tagging.

* Create a archive file for test on local.

  ```console
  $ rake dist
  ```

* Change `~/.dput.cf` in order to upload the `nightly` repository.

>   Add or change a `[groonga-ppa]` entry as below.

Author: Fixed.

  `incoming = ~groonga/ubuntu/nightly` is important.

  If you don't have `~/.dput.cf`, please manually create it.

  ```console
  $ vi ~/.dput.cf
  [groonga-ppa]
  fqdn = ppa.launchpad.net
  method = ftp
  incoming = ~groonga/ubuntu/nightly
  login = anonymous
  allow_unsigned_uploads = 0
  ```

* Upload to the `nightly` repository.

  ```console
  $ rake package:ubuntu
  ```

* Check the build result

  When upload packages into the `nightly` reposigoty succeeded, 
  a package build process is executed on launchpad.net.

>   Then build result is notified via E-mail.

Reviewer: LaunchPad send build result with E-mail to us only when the package build fail.

Author: ビルドが失敗したときのみ送信する、という文言に変更しました。

>   Then build result is notified via E-mail.

Author: Fixed

## Tagging for the release

```console
$ rake tag
```

## Download or create a archive file

Donwload the archive file (`pgroonga-x.x.x.tar.gz`) from the 

> [GitHub relase page](https://github.com/pgroonga/pgroonga/releases/latest)

Author: Fixed

> and distribute it to the top of the PGroonga repository.

Author: distributeという単語が良くなかった気がするのですが、ここでは自分のローカルにcloneしてあるPGroongaのリポジトリーのことを指していました。
（この手順を実施するときは、前提条件としてPGroongaのリポジトリーは手元にcloneしてあります）

distributeをmoveくらいに変更して、 the PGroonga repositoryを `your local PGroonga repository` に変更します。

> and distribute it to the top of the PGroonga repository.

Reviewer 2: working directory or working tree?
https://git-scm.com/book/en/v2/Git-Tools-Reset-Demystified

> and distribute it to the top of the PGroonga repository.

Author: working directoryを使うようにしました。

If the archive file doesn't exist on the GitHub relase page, 

> you can create it on local by the following command.

Author: ここで案内している `rake dist` を実行すると、ローカルのPGroongaリポジトリのトップにアーカイブファイル (`pgroonga-x.x.x.tar.gz`) ができるので、移動する必要がないんです。
ただ、ローカルで作ると環境差異などで、上手く作れなかったり想定外のものができたりする可能性があるので、GitHub release page にあるものをダウンロードしてローカルのPGroongaリポジトリのトップに移動する手順を案内しています。

> you can create it on local by the following command.

Reviewer 3: なるほど。それであれば " create it in a top of the local repository" とかでもよかったかもですね。
create on local だとローカルの上にというかんじなのでローカルリポジトリのトップに作れるというよりはローカルに新たにリポジトリをドーンとつくる。という印象になるとおもいます。

> you can create it on local by the following command.

Author: なるほど、ありがとうございます。

https://github.com/pgroonga/pgroonga.github.io/pull/89#discussion_r1025878564

でそもそもこの分自体を削除するのが良さそうということになったのですが、表現のニュアンスの違いを覚えておきます。

`nightly` is for testing, and `ppa` is for distributing.

We should test whether we can build packages for Ubuntu on the `nightly` repository before tagging.

* Create a archive file for test on local.

  ```console
  $ rake dist
  ```

* Change `~/.dput.cf` in order to upload the `nightly` repository.

  Add or change a `[groonga-ppa]` entry as below.

  ```console
  $ vi ~/.dput.cf
  [groonga-ppa]
  fqdn = ppa.launchpad.net
  method = ftp
  incoming = ~groonga/ubuntu/nightly
  login = anonymous
  allow_unsigned_uploads = 0
  ```

  `incoming = ~groonga/ubuntu/nightly` is important.

  If you don't have `~/.dput.cf`, create it manually.

* Upload to the `nightly` repository.

  ```console
  $ rake package:ubuntu
  ```

* Check the build result

 When uploading packages into the `nightly` repository is succeeded, 
 a package build process is executed on launchpad.net.
 Then build result is notified via E-mail if the build fails.

## Tagging for the release

```console
$ rake tag
```

## Download or create a archive file

Donwload the archive file (`pgroonga-x.x.x.tar.gz`) from the 
[GitHub release page](https://github.com/pgroonga/pgroonga/releases/latest)

> and move it to the top of your local PGroonga repository.

Reviewer 2: 現状でアーカイブをアップロードするCIってあります？
Groongaみたいに`.github/workflow/package.yml`からソースを生成しているジョブを分離してリリースできるようにするのを先にやらないといけなくないっすか？

Author: ご指摘の通り、現状はありません。
今回のリリースまでに作成する間に合わせるつもりでいたので、記載していました。
また、間に合わなかった場合も考えて、GitHubリリースページ上になければローカルで作る旨を追加していました。

Reviewer 2: 30分とか1時間くらいでちょろっとできると思うので用意してしまってドキュメントを減らしたほうがよくないっすか？

Author: そのようにします！

$ rake yum
```

### Update Docker images

> We update Docker images of [Docker Hub][pgroonga-docker-hub].

Reviewer: ```suggestion
We update PGroonga's Docker images of [Docker Hub][pgroonga-docker-hub].
```

> Clone [Docker repository][pgroonga-docker-repository] and update Dockerfiles.

Reviewer: ```suggestion
Clone [PGroonga's Docker repository][pgroonga-docker-repository] and update Dockerfiles.
```

Author: `Dockerfiles`  appears first time in this description. So I think `the` is needless.

Author: ```
Clone [PGroonga's Docker repository][pgroonga-docker-repository] and update Dockerfiles in tha repository.
```

I have changed this sentence as above.

Here is an example for the case that the PGroonga version is `2.4.1`, and the Groonga version is `12.0.9`.

```
$ mkdir -p ~/work/pgroonga
$ rm -rf ~/work/pgroonga/docker.clean
$ git clone --recursive git@github.com:pgroonga/docker.git ~/work/pgroonga/docker.clean
$ cd ~/work/pgroonga/docker.clean
$ ./update.sh 2.4.1 12.0.9 #Automatically update Dockerfiles and commit changes and create a tag.
$ git push
```

You have to specify the latest versions.

> [GitHub Actions of Docker repository][github-actions-pgroonga-docker] automatically update the Docker images of [Docker Hub][pgroonga-docker-hub] after you push the change.

Reviewer: ```suggestion
[GitHub Actions of PGroonga's Docker repository][github-actions-pgroonga-docker] automatically update the docker images of [PGroonga's Docker Hub][pgroonga-docker-hub] after you push the change.
```

## Announce release

### Announce release for mailing list

We send release announcement to PostgreSQL mailing list if we have big news.

Your PostgreSQL account must belong to `PGroonga project` for announcement.
Please contact a project member in order to join `PGroonga project`.

* https://www.postgresql.org/list/

> * https://www.postgresql.org/search/?m=1&ln=pgsql-announce&q=PGroonga (Archives for PGroonga announcements)

Reviewer: ```suggestion
* [https://www.postgresql.org/search/?m=1&ln=pgsql-announce&q=PGroonga](https://www.postgresql.org/search/?m=1&ln=pgsql-announce&q=PGroonga) (Archives for PGroonga announcements)
```

### Announce release for mailing list

> We send release announce to PostgreSQL mailing list if we have big news.

Reviewer 3: ```suggestion
We send release announcement to PostgreSQL mailing list if we have big news.
```
announceでもよいですが次の文章でannouncementになっているので。(announcementのほうが正確には名詞です）

Your PostgreSQL account must belong to `PGroonga project` for announcement.

> Please contact other release personnel to belong to `PGroonga project`.

Reviewer 3: ```suggestion
Please contact other release personnel in order to join `PGroonga project`.
```
所属するために問い合わせてほしいという事であれば　「入るために」でよいと思います。
一方でother release personnelよりはproject memberとかのほうがよいのでは？ともおもいました。

* https://www.postgresql.org/list/
* https://www.postgresql.org/search/?m=1&ln=pgsql-announce&q=PGroonga (Archives for PGroonga announcements)

> ### Announce release for GitHub Discuttions

Reviewer: ```suggestion
### Announce release for GitHub Discussions
```

> We make release announce in GitHub Discussions.

Reviewer 3: ```suggestion
We make release announcement in GitHub Discussions.
```
